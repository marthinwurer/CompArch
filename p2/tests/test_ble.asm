;
; tests BLE
;
; uses instructions MOV, BPT, BR, HALT, ADD, and CMP
;
; uses addressing modes 0, 2, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	177777
v3:	.word	077777
;
start:	mov	v1,r1
	mov	v2,r2
	mov	v3,r3
	bpt
;
	cmp	r1,r3
	ble	ok1
	add	#1,r0
ok1:
;
	cmp	r3,r1
	ble	err2
	br	ok2
err2:	add	#2,r0
ok2:
;
	cmp	r1,r2
	ble	err3
	br	ok3
err3:	add	#4,r0
ok3:
;
	cmp	r2,r1
	ble	ok4
	add	#10,r0
ok4:
;
	cmp	r3,r2
	ble	err5
	br	ok5
err5:	add	#20,r0
ok5:
;
	cmp	r2,r3
	ble	ok6
	add	#20,r0
ok6:
;
	cmp	r1,r1
	ble	ok7
	add	#40,r0
ok7:
;
	bpt
	halt
;
	.end	start
