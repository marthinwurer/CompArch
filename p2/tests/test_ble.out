CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016701 177766  MOV 177766(R7),R1
000012:  PS[04]  016702 177764  MOV 177764(R7),R2
000016:  PS[10]  016703 177762  MOV 177762(R7),R3
000022:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000024]
000024:  PS[00]  020103  CMP R1,R3
000026:  PS[11]  003402  BLE 4
000034:  PS[11]  020301  CMP R3,R1
000036:  PS[00]  003401  BLE 2
000040:  PS[00]  000402  BR 4
000046:  PS[00]  020102  CMP R1,R2
000050:  PS[01]  003401  BLE 2
000052:  PS[01]  000402  BR 4
000060:  PS[01]  020201  CMP R2,R1
000062:  PS[10]  003402  BLE 4
000070:  PS[10]  020302  CMP R3,R2
000072:  PS[13]  003401  BLE 2
000074:  PS[13]  000402  BR 4
000102:  PS[13]  020203  CMP R2,R3
000104:  PS[10]  003402  BLE 4
000112:  PS[10]  020101  CMP R1,R1
000114:  PS[04]  003402  BLE 4
000122:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000124]
000124:  PS[04]  000000  HALT

Machine Halted - HALT instruction

Simulated time 162 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
